packages/components/src/index.ts

```diff
@@ -116,12 +116,33 @@ export { default as __experimentalNavigationGroup } from './navigation/group';
 export { default as __experimentalNavigationItem } from './navigation/item';
 export { default as __experimentalNavigationMenu } from './navigation/menu';
 export {
+	/**
+	 * @deprecated Import `Navigator` instead.
+	 */
 	NavigatorProvider as __experimentalNavigatorProvider,
+	/**
+	 * @deprecated Import `Navigator` and use `Navigator.Screen instead.
+	 */
 	NavigatorScreen as __experimentalNavigatorScreen,
+	/**
+	 * @deprecated Import `Navigator` and use `Navigator.Button` instead.
+	 */
 	NavigatorButton as __experimentalNavigatorButton,
+	/**
+	 * @deprecated Import `Navigator` and use `Navigator.BackButton` instead.
+	 */
 	NavigatorBackButton as __experimentalNavigatorBackButton,
+	/**
+	 * @deprecated Import `Navigator` and use `Navigator.ToParentButton` instead.
+	 */
 	NavigatorToParentButton as __experimentalNavigatorToParentButton,
+	/**
+	 * @deprecated Import `useNavigator` instead.
+	 */
 	useNavigator as __experimentalUseNavigator,
+	NavigatorProvider,
+	Navigator, // alias for `NavigatorProvider`
```

Comment:
I'm not sure if this is a good idea, but I wanted to try it out and see what you think of it. The gist is for the root component - the main namespace for the other children components to be simply called `Navigator`:

```jsx
import {
  Navigator,
} from '@wordpress/components';

const MyNavigation = () => (
  <Navigator initialPath="/">
    <Navigator.Screen path="/">
      <p>This is the home screen.</p>
       <Navigator.Button path="/child">
         Navigate to child screen.
      </Navigator.Button>
    </Navigator.Screen>

    <Navigator.Screen path="/child">
      <p>This is the child screen.</p>
      <Navigator.ToParentButton>
        Go back
      </Navigator.ToParentButton>
    </Navigator.Screen>
  </Navigator>
);
```

Pros: The name fits better semantically. It's shorter, too :)

Cons: This means we'll need to keep re-exporting `NavigatorProvider` here for backward compatibility, though. Exporting both might create confusion. Also, the docs need to reference both, which is not ideal. You could also argue that it's better to be explicit about it being a provider.

We can easily revert if we judge it to be too much trouble for its worth.



I thi

Reply:
Will leave a comment directly in the main PR's conversation